mt_dnn/batcher.py

```diff
@@ -34,6 +53,27 @@ def _get_shuffled_index_batches(dataset_len, batch_size):
         random.shuffle(index_batches)
         return index_batches
 
+    @staticmethod
+    def _get_shuffled_index_batches_bin(dataset, batch_size, bin_size, bin_grow_ratio):
+        maxlen = dataset._maxlen
+        bins = create_bins(bin_size, maxlen)
+        data = [[] for i in range(0, len(bins))]
+        
+        for idx, sample in enumerate(dataset):
+            bin_idx = search_bin(bins, len(sample['sample']['token_id']))
+            data[bin_idx].append(idx)
+        index_batches = []
+
+        for idx, sub_data in enumerate(data):
```

Comment:
the length of bins are [64, 128, 192, 256, 320, 384, 448, 512], which increases linearly,
but the batch_size scales down exponentially by "batch_size * bin_grow_ratio", is this the optimal?
I guess the memory grows linearly with O(len) , then could batch_size = batch_size_for_64 // cur_bin_seq_len?

Reply:
if O(len^2) batch_size = batch_size_for_64 // (128*128// (64*64))